common/taskana-common/src/main/java/pro/taskana/common/internal/test/InitOracleSchema.java

```diff
@@ -0,0 +1,70 @@
+package pro.taskana.common.internal.test;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.Statement;
+import javax.sql.DataSource;
+
+import pro.taskana.common.api.exceptions.SystemException;
+
+/**
+ * Helper Class to create a proper ORACLE SCHEMA
+ *
+ * <p>This class is only used in TEST classes. DON'T USE IT IN PRODUCTION CODE.
+ */
+public final class InitOracleSchema {
+
+  public static void initOracleSchema(DataSource dataSource, String schemaName)
+      throws SystemException {
+    try (Connection connection = dataSource.getConnection()) {
+      Connection conn =
+          DriverManager.getConnection(connection.getMetaData().getURL(), "SYSTEM", "testPassword");
+      try (Statement stmt = conn.createStatement()) {
+        stmt.execute("GRANT ALL PRIVILEGES TO TEST_USER");
+
+        if ("TASKANA".equals(schemaName)) {
+          try {
+            stmt.execute("DROP USER " + schemaName + " CASCADE");
+            stmt.execute(
+                "DROP TABLESPACE "
+                    + schemaName
+                    + "_TMP INCLUDING CONTENTS AND DATAFILES CASCADE CONSTRAINTS");
+            stmt.execute(
+                "DROP TABLESPACE "
+                    + schemaName
+                    + " INCLUDING CONTENTS AND DATAFILES CASCADE CONSTRAINTS");
+          } catch (Throwable e) {
+            // IGNORE exceptions from dropping
+          }
+        }
+
+        stmt.execute(
+            "create tablespace "
+                + schemaName
+                + " datafile '"
+                + schemaName
+                + ".dat' size 5M autoextend on NEXT 5M MAXSIZE UNLIMITED");
+        stmt.execute(
+            "create temporary tablespace "
+                + schemaName
+                + "_TMP tempfile '"
+                + schemaName
+                + "_tmp.dat' size 5M autoextend on NEXT 5M MAXSIZE UNLIMITED");
+        stmt.execute(
+            "create user "
+                + schemaName
+                + " identified by testPassword default tablespace "
+                + schemaName
+                + " temporary tablespace "
+                + schemaName
+                + "_TMP");
+        stmt.execute("ALTER USER " + schemaName + " quota unlimited on " + schemaName);
+        stmt.execute("GRANT UNLIMITED TABLESPACE TO " + schemaName);
```

Comment:
I have no idea what's happening here. Should I or is this something oracle specific?
I remember that the purpose of this is to create a test user which has access to all schemas (generated through the test-api)

Reply:
In oracle a schema is not simple a schema

a schema is a user and a user/schema must have tablespace and temporary tablespace

therefore i create 
one tablespace
one temporary tablespace
and one user which "owns" the two new tablespaces

the drop user and tablespace is only for the legacy part and normaly not needed